### PR TITLE
Lambda runtime parity

### PR DESCRIPTION
**Pull request by joe4dev**
Improves lambda runtime parity related to user and environment variables:

* Update Lambda runtime RIE:
  * Switch to `sbx_user1051` user for runtime parity and drop root privileges.
  * Unset internal environment variables
* Systematically group environment variables fix parity mismatches
* Extend snapshot tests for important paths regarding permissions and user/group attributes
* Add `LAMBDA_INIT_USER` flag and populate it automatically to overcome RIE debugging limitation when dropping privileges

Related to https://github.com/localstack/lambda-runtime-init/pull/13

Addresses https://github.com/localstack/localstack/issues/7722

### CODE REVIEW

File: localstack/services/awslambda/packages.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 LAMBDA_RUNTIME_INIT_URL = "https://github.com/localstack/lambda-runtime-init/releases/download/{version}/aws-lambda-rie-{arch}"
 
-LAMBDA_RUNTIME_DEFAULT_VERSION = "v0.1.11-pre"
+LAMBDA_RUNTIME_DEFAULT_VERSION = "v0.1.12-pre"
```
Comment (joe4dev) [inline localstack/services/awslambda/packages.py:12]: This is the key change adopting the new RIE release based with https://github.com/localstack/lambda-runtime-init/pull/13/files

File: localstack/services/awslambda/invocation/runtime_environment.py [hunk 2]
```diff
@@ -87,38 +87,64 @@ def get_environment_variables(self) -> Dict[str, str]:
...
             "LOCALSTACK_HOSTNAME": self.runtime_executor.get_endpoint_from_executor(),
             "EDGE_PORT": str(config.EDGE_PORT),
             "AWS_ENDPOINT_URL": f"http://{self.runtime_executor.get_endpoint_from_executor()}:{config.EDGE_PORT}",
+            # 4) Internal LocalStack runtime API
+            "LOCALSTACK_RUNTIME_ID": self.id,
+            "LOCALSTACK_RUNTIME_ENDPOINT": self.runtime_executor.get_runtime_endpoint(),
+            # LOCALSTACK_USER conditionally added below
         }
+        # Conditionally added environment variables
+        # TODO: Can handler be None?
```
Comment (joe4dev) [inline localstack/services/awslambda/invocation/runtime_environment.py:133]: Can handler be None?

Comment (dfangl) [reply localstack/services/awslambda/invocation/runtime_environment.py:133]: In image lambdas, yes

**Review by dominikschubert — APPROVED** [review]
LGTM :rocket: 

Good idea to reorder the env variables by context, rather than purely alphabetically :+1:

File: localstack/services/awslambda/invocation/runtime_environment.py (hunk 2 above)
Comment (joe4dev) [reply localstack/services/awslambda/invocation/runtime_environment.py]: Right, at API level `Handler` is only required for zip lambdas. I added a clarifying comment.

The `_HANDLER` environment variable is then populated at runtime (e.g., by RIE). Examples:
* `/application` for the aws-samples app [lambda-from-container](https://github.com/aws-samples/aws-cdk-examples/blob/master/python/lambda-from-container)
* `app.handler` in our ext test `tests.integration.test_lambda.TestLambdaContainer.test_lambda_from_image`


Sidenote: CDK validates it `TypeError: Function.__init__() missing 1 required keyword-only argument: 'handler'` (see example [here](https://github.com/aws-samples/aws-cdk-examples/blob/master/python/lambda-from-container/app.py#L73))

**Pull request merged**